### ISSUE TITLE
All my projects directory is cleared after error!!!

### ISSUE THREAD

**Issue by imkrimerman** [ApiGen/ApiGen#550]
After Error (29%ERROR: application encountered an error and can not continue. Error was logged.) while generating documentation, folder with my projects is empty, only some html documentation files. It's very awful. I've worked for 3 days on new features.

Destination folder was my projects folder.

**Comment by TomasVotruba** [comment]
I'm sorry about that. It never occurred to me to use this option to some global directory, because it would be full of .html files. Dir is cleaned to prevent storing some old files.

Minimal solution would be add not to command option or if you have some suggestion.

Proper solution would be to implement diff logic (e.g. https://github.com/FriendsOfPHP/Sami/blob/master/Sami/Renderer/Diff.php, https://github.com/FriendsOfPHP/Sami/blob/8b13ef74b8296b660cef1dbd757a9b47bb81314d/Sami/Renderer/Renderer.php#L48), that would render diff and throw the rest of related files.

**Comment by fprochazka** [comment]
@imkrimerman let this be a lesson to you... use git and push regularly.

I think the proper solution would be to ask the user if he want's to clean the directory before generating the docs, if there are contents in it.

**Comment by TomasVotruba** [comment]
@fprochazka Good idea simple to implement: add [question helper](http://symfony.com/doc/current/components/console/helpers/questionhelper.html#asking-the-user-for-confirmation) to [GenerateCommand](https://github.com/ApiGen/ApiGen/blob/cc1a49d5046ae0ec822036899c05738f6ecbd42b/src/Command/GenerateCommand.php#L173).

Feel free to send PR, if anyone has some time.

---
Still I'd prefer not to bother user if not needed. Maybe as next step.

**Comment by fprochazka** [comment]
@TomasVotruba that's exactly what I had in mind :)

**Comment by sandio** [comment]
Hi, I recently updated ApiGen from 2.8.1 to 4.1. We use ApiGen with our build server. 

Do you consider a good idea to have a flag that will _skip_ asking to wipe destination, if not empty. The alternative for us is to wipe the destination manually or pass 'yes' to ApiGen to wipe it for us.

**Comment by TomasVotruba** [comment]
Hi, try `-q` for quiet mode.

**Comment by sandio** [comment]
Hi, I tried it. ApiGen is still waiting for user input in quiet mode. Only when I type yes/no or hit Return, it continues generating.

Example:
`apigen generate --source src/ --destination apigen/ --quiet`

I wanted to ask what happens when I type _no_ as an answer to the question whether to erase the destination. As of user point of view, the console output remains the same. I looked at the commit that introduced this feature (4ef7366) and think ApiGen overwrites the destination folder. Is this correct?

Thank you.
